jwst/residual_fringe/residual_fringe_step.py

```diff
@@ -0,0 +1,131 @@
+#! /usr/bin/env python
+from ..stpipe import Step
+from .. import datamodels
+from . import residual_fringe
+from functools import partial
+
+__all__ = ["ResidualFringeStep"]
+
+
+class ResidualFringeStep(Step):
+    """
+    ResidualFringeStep: Apply residual fringe correction to a science image using a residual fringe
+    reference image.
+
+    Parameters
+    ----------
+    input_data : asn file or ModelContainer
+        Single filename association table or a datamodels.ModelContainer
+    """
+
+    spec = """
+        save_intermediate_results  = boolean(default = False)
+        transmission_level = integer(default=80) # transmission level to use to define slice locations
+        search_output_file = boolean(default = False)
+    """
+
+    reference_file_types = ['fringefreq','regions']
+
+    def process(self, input):
+
+        self.suffix = 'residual_fringe'
```

Comment:
This should be done in the `spec`. Add the following into the spec:

`suffix = string(default="residual_fringe")`